homeassistant/components/generic/camera.py

```diff
@@ -75,25 +75,74 @@ async def async_setup_platform(
 ) -> None:
     """Set up a generic IP Camera."""
 
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+    _LOGGER.warning(
+        "Loading generic IP camera via configuration.yaml is deprecated, "
+        "it will be automatically imported.  Once you have confirmed correct "
+        "operation, please remove 'generic' (IP camera) section(s) from "
+        "configuration.yaml"
+    )
+    image = config.get(CONF_STILL_IMAGE_URL)
+    stream = config.get(CONF_STREAM_SOURCE)
+    config_new = {
+        CONF_NAME: config[CONF_NAME],
+        CONF_STILL_IMAGE_URL: image.template if image is not None else None,
+        CONF_STREAM_SOURCE: stream.template if stream is not None else None,
+        CONF_AUTHENTICATION: config.get(CONF_AUTHENTICATION),
+        CONF_USERNAME: config.get(CONF_USERNAME),
+        CONF_PASSWORD: config.get(CONF_PASSWORD),
+        CONF_LIMIT_REFETCH_TO_URL_CHANGE: config.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE),
+        CONF_CONTENT_TYPE: config.get(CONF_CONTENT_TYPE),
+        CONF_FRAMERATE: config.get(CONF_FRAMERATE),
+        CONF_VERIFY_SSL: config.get(CONF_VERIFY_SSL),
+    }
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config_new
+        )
+    )
+
+
+async def async_setup_entry(hass, config, async_add_entities):
+    """Set up a generic IP Camera."""
 
-    async_add_entities([GenericCamera(hass, config)])
+    async_add_entities([GenericCamera(hass, config.options, config.unique_id)])
+
+
+def generate_auth(device_info) -> httpx.Auth | None:
+    """Generate httpx.Auth object from credentials."""
+    username = device_info.get(CONF_USERNAME)
+    password = device_info.get(CONF_PASSWORD)
+    authentication = device_info.get(CONF_AUTHENTICATION)
+    if username:
+        if authentication == HTTP_DIGEST_AUTHENTICATION:
+            return httpx.DigestAuth(username=username, password=password)
+        return httpx.BasicAuth(username=username, password=password)
+    return None
 
 
 class GenericCamera(Camera):
     """A generic implementation of an IP camera."""
 
-    def __init__(self, hass, device_info):
+    def __init__(self, hass, device_info, identifier):
         """Initialize a generic camera."""
         super().__init__()
         self.hass = hass
+        self._attr_unique_id = identifier
         self._authentication = device_info.get(CONF_AUTHENTICATION)
         self._name = device_info.get(CONF_NAME)
```

Comment:
```suggestion
        self._name = device_info.get(CONF_NAME, entry.title)
```

If name isn't set you can set it to `entry.title`

Reply:
I don't think this works... `entry` is not defined here.

Reply:
You'll have to pass it in above